src/transaction/txParser.ts

```diff
@@ -11,89 +25,176 @@ import {
+      throw Error(Errors.TxMultiHostNameRelayParseError)
+    }
+    const [type, dnsName] = relay
+    return {
+      type,
+      dnsName,
+    }
+  }
+  switch (poolRelay[0]) {
+    case TxRelayTypes.SINGLE_HOST_IP:
+      return parseSingleHostIPRelay(poolRelay)
+    case TxRelayTypes.SINGLE_HOST_NAME:
+      return parseSingleHostNameRelay(poolRelay)
+    case TxRelayTypes.MULTI_HOST_NAME:
+      return parseMultiHostNameRelay(poolRelay)
+    default:
+      throw Error(Errors.UnsupportedRelayTypeError)
+  }
+}
+
+const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   const stakeKeyRegistrationCertParser = (
-    [type, [, pubKeyHash]]: any,
-  ): _StakingKeyRegistrationCert => ({ type, pubKeyHash })
+    txCertificate: any,
+  ): _StakingKeyRegistrationCert => {
+    if (!isTxStakingKeyRegistrationCert(txCertificate)) {
+      throw Error(Errors.TxStakingKeyRegistrationCertParseError)
+    }
+    const [type, [, pubKeyHash]] = txCertificate
+    return ({ type, pubKeyHash })
+  }
 
   const stakeKeyDeregistrationCertParser = (
-    [type, [, pubKeyHash]]: any,
-  ): _StakingKeyDeregistrationCert => ({ type, pubKeyHash })
+    txCertificate: any,
+  ): _StakingKeyDeregistrationCert => {
+    if (!isStakingKeyDeregistrationCert(txCertificate)) {
+      throw Error(Errors.TxStakingKeyDeregistrationCertParseError)
+    }
+    const [type, [, pubKeyHash]] = txCertificate
+    return ({ type, pubKeyHash })
+  }
 
   const delegationCertParser = (
-    [type, [, pubKeyHash], poolHash]: any,
-  ): _DelegationCert => ({ type, pubKeyHash, poolHash })
+    txCertificate: any,
+  ): _DelegationCert => {
+    if (!isDelegationCert(txCertificate)) {
+      throw Error(Errors.TxDelegationCertParseError)
+    }
+    const [type, [, pubKeyHash], poolHash] = txCertificate
+    return ({ type, pubKeyHash, poolHash })
+  }
 
   const stakepoolRegistrationCertParser = (
-    [
+    txCertificate: any,
+  ): _StakepoolRegistrationCert => {
+    if (!isStakepoolRegistrationCert(txCertificate)) {
+      throw Error(Errors.TxStakepoolRegistrationCertParseError)
+    }
+    const [
       type,
-      poolPubKey,
-      operatorPubKey,
-      fixedCost,
-      margin,
-      tagged,
-      rewardAddressBuff,
-      ownerPubKeys,
-      s1,
-      s2,
-    ]: any,
-  ): _StakepoolRegistrationCert => ({
-    // TODO: check whether this is accurate and which of these we actually need{
-    type,
-    poolPubKey,
-    operatorPubKey,
-    fixedCost,
-    margin,
-    tagged,
-    rewardAddressBuff,
-    ownerPubKeys,
-    s1,
-    s2,
-  })
-
-  type certficateParser =
-    | typeof stakeKeyRegistrationCertParser
-    | typeof stakeKeyDeregistrationCertParser
-    | typeof delegationCertParser
-    | typeof stakepoolRegistrationCertParser
+      poolKeyHash,
+      vrfPubKeyHash,
+      pledge,
+      cost,
+      { value },
+      rewardAddress,
+      poolOwnersPubKeyHashes,
+      relays,
+      [metadataUrl, metadataHash],
```

Comment:
Same as in the other comment, metadata can be legit null and this can fail on destructuring. These are the places it failed during adalite integration, other parts of the code can be affected as well